verilog/sim_two_power_mod.cpp

```diff
@@ -0,0 +1,73 @@
+
+#include "Vtwo_power_mod.h"
+#include "callback.h"
+#include "dut_wrapper.h"
+#include "verilog_int.h"
+#include <iostream>
+#include <systemc>
+
+using namespace std;
+using namespace sc_core;
+
+using DUT = Vtwo_power_mod;
+using KeyType = verilog::vint<false, 256>;
+struct RSATwoPowerModIn {
+  friend ::std::ostream &operator<<(::std::ostream &os,
+                                    const RSATwoPowerModIn &v) {
+    os << "{" << v.power << ", " << v.modulus << "}" << std::endl;
+    return os;
+  }
+
+  sc_dt::sc_uint<32> power;
+  KeyType modulus;
+};
+using RSATwoPowerModOut = KeyType;
+
+const char str_modulus[] =
+    "0xE07122F2A4A9E81141ADE518A2CD7574DCB67060B005E24665EF532E0CCA73E1";
+const char str_ans[] =
+    "0AF39E1F831CB4FCD92B17F61F473735C687593A931C97D2B60AD6C7443F09FDB";
+const int power = 512;
+
+SC_MODULE(Testbench) {
+  sc_clock clk;
+  DUTWrapper<DUT> dut_wrapper;
+  shared_ptr<Driver<RSATwoPowerModIn>> driver;
+  shared_ptr<Monitor<RSATwoPowerModOut>> monitor;
+
+public:
+  SC_HAS_PROCESS(Testbench);
+  Testbench(const sc_module_name &name)
+      : sc_module(name), clk("clk", 1.0, SC_NS), dut_wrapper("dut_wrapper") {
+    dut_wrapper.clk(clk);
+    driver = make_shared<Driver<RSATwoPowerModIn>>(
+        dut_wrapper.dut->i_valid, dut_wrapper.dut->i_ready,
+        [=](RSATwoPowerModIn in) { this->writer(in); });
+    monitor = make_shared<Monitor<RSATwoPowerModOut>>(
+        dut_wrapper.dut->o_valid, dut_wrapper.dut->o_ready,
+        [=]() { return this->reader(); });
```

Comment:
You do not use the return value, why?

Reply:
The monitor can decide what to do with the return value of the data, I think? The reader just return the value to Monitor.